### ISSUE TITLE
The Debian CD image does not boot on Miix 630

### ISSUE THREAD

**Issue by ajalexei** [aarch64-laptops/debian-cdimage#7]
The current Debian CD image -- debian-unstable-arm64-DVD-1_20210104.iso -- does not boot on Miix 630 (secure boot disabled, bitlocker turned off). I do not see any grub promt even -- blank screen for a while and booting Windows afterwards. I presume it is not the issue with GRUB mentioned [here](https://www.linaro.org/blog/porting-linux-to-aarch64-laptops/): 

> The laptops implement UEFI support so we disabled Secure Boot, inserted a bootable SD card containing Grub for AArch64 and … nothing. No error message. Not a sausage! Just a black screen. Thankfully, after a long period of head scratching some bright soul discovered that a bug in the firmware requires the bootable binary, Grub in this case, to be compiled 4k aligned for the firmware to take any notice of it. What we saw (or were not seeing) was UEFI Boot Services rejecting Grub as a viable binary.

Also the usb stick wih the CD image seems to be corrupted after the boot attempt -- I fail to mount the EFI partition under linux.

Any suggestions what went wrong? I am happy to provide more information or to make extra tests.

**Comment by shawnguo2** [comment]
Hmm, although I haven't tested the CD image on Miix 630 and I doubt it will just work there, I would expect that you can at least get GRUB prompt from USB disk booting.  And it shouldn't be the problem of 4k alignment, as it's been fixed in grub-2.04-rc1.

The corruption of USB stick sounds interesting, as booting from there is all about reading stuff and never writing.  Were you able to mount the EFI partition before the boot attempt?

**Comment by ajalexei** [comment]
Yes, I was able to properly mount both partitions on the stick prior to the boot attempt. Further experiments show that it is the problem with the stick rather than the image: a different USB stick boots into grub and is not corrupted afterwards -- I didn't have time to go any further today -- hopefully I will over the weekend. I expect the next trouble to be the missing DTB for Miix 630.

**Comment by shawnguo2** [comment]
The next thing would be to see if the [installer kernel](https://github.com/aarch64-laptops/linux/commits/laptops-5.10) can boot Miix 630 from ACPI or not.  Missing DTB will only become a problem after you finish install and try to boot from UFS.

**Comment by ajalexei** [comment]
Thanks. Indeed the troubles started before the DTB: I think the installer kernel starts booting (see the attached images -- sorry for the poor quality: the messages only showed up for a fraction of a second) before the screen goes blank and nothing seem to be happening anymore.

![P1280368 (2)](https://user-images.githubusercontent.com/23712361/106154905-74bd1d80-61c3-11eb-8f20-3ca47bfcc78f.JPG)
![P1280377 (2)](https://user-images.githubusercontent.com/23712361/106154918-771f7780-61c3-11eb-89a1-7fadc664ea71.JPG)

**Comment by shawnguo2** [comment]
It looks there are issues with ACPI boot on Snapdragon 835, and some debugging effort is needed.  But unfortunately, I do not have any Snapdragon 835 based laptops to look into it.

**Comment by ajalexei** [comment]
Thanks. Too bad -- there does not seem to be any reasonably cheap Snapdragons 835 on ebay around this time.
My understanding is that one can still boot using the DTB, like the original aarch64-laptop project Ubuntu image does.
Is there a way to modify the debian image to do that?
I could also try to do the debugging, but frankly my experience in this area is zero.
